rfcs/0011-removal-managed-zone.md

```diff
@@ -0,0 +1,97 @@
+
+# Motivation
+[motivation]: #motivation
+
+The `ManagedZone` API is required in order to create a DNSPolicy. Its original purpose and value is no longer present. It now creates an additional barrier for setting up a `DNSPolicy` that was never intended or really needed once we moved away from offering dns operator as a service. We want to improve this user experience while also simplifying our object model and controller code base.
+
+The key user impact here is to avoid users having to create additional none policy `Kuadrant APIs` in order to be able to leverage the `DNSPolicy` API. 
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+You will no longer need or be able to create a `ManagedZone` resource. It will be expected that the zone already exists in your provider of choice. Now you will only create a credential via a Kubernetes secret. This secret will be where any additional configuration is also applied such as the zone id to target and the root domain in that zone to use. Removing the managed zone API will reduce API calls to the DNS provider and reduce events and load on the server and kuberentes API server.
+You will no longer be limited to a single root domain or single provider per policy. The managed zone was a single reference, however we want to move to leverage a label selector which will mean multiple credentials can be used for a single policy. 
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The `ManagedZone` API will be fully removed. All tests will be updated/removed as needed. We will rename the `managedZoneRef` property to be `providerSelector` that will be a required field and will accept a label selector for secrets within the same namespace as the policy resource. We will expect the zone id and root domain to be specified in the provider secret, these will be required. These zone ids will limit what zones the controller will interact with and manage records within. 
+
+## Secret structure
+
+under the data field we will expect two new fields to be present
+
+```
+
+ZONE_ID: somezone
+DOMAIN_NAME: *.a.b.com
+
+```
+
+If these fields are not present the status of the DNSPolicy will be updated to reflect the issue (more on status below). Only if these fields are present and listeners that match the `DOMAIN_NAME` are present will DNSRecords be created. If no record is created for a given secret, this will be considered a valid configuration (listeners may be added after a secret has been created and we shouldn't assume an error in this case).
+
+
+## Secret label selector
+
+The reason for using a label selector rather than a direct reference are as follows:
```

Comment:
I am missing reasoning why label selector are better than allowing multiple credential reference directly (i.e. credentials being a list of references to secrets).

Personally, I think this particular use case is better suited for the direct specification in the policy, than to have dynamic secret selectors (it is easier to work with as a user, easier to validate, everything can be seen from the spec alone)

Reply:
I think this is worth thinking about. We have also been debating that. Label selectors are very common method for allow mutliple objects to be referenced, however a list type would also be an option.

Reply:
How about writing it as an alternative solution in the RFC, properly exploring and investigating pros and cons of each solution

Reply:
This has been updated now and uses and array